controllers/aggregated_clusteroperator_controller.go

```diff
@@ -62,34 +63,40 @@ func (a *AggregatedClusterOperatorReconciler) Reconcile(ctx context.Context, req
 	}
 	defer func() {
 		if err := coWriter.UpdateStatus(ctx, aggregatedCO, coBuilder.GetStatus()); err != nil {
-			log.Error(err, "error updating CO status")
+			log.Error(err, "error updating cluster operator status")
 		}
 	}()
 
 	// Set the default CO status conditions: Progressing=True, Degraded=False, Available=False
-	coBuilder.WithProgressing(openshiftconfigv1.ConditionTrue, "")
-	coBuilder.WithDegraded(openshiftconfigv1.ConditionFalse)
-	coBuilder.WithAvailable(openshiftconfigv1.ConditionFalse, "", "")
+	// TODO: always set a reason (message is optional, but desirable)
+	coBuilder.WithProgressing(metav1.ConditionTrue, "", "")
+	coBuilder.WithDegraded(metav1.ConditionFalse, "", "")
+	coBuilder.WithAvailable(metav1.ConditionFalse, "", "")
 
 	poList := &platformv1alpha1.PlatformOperatorList{}
 	if err := a.List(ctx, poList); err != nil {
 		return ctrl.Result{}, err
 	}
 	if len(poList.Items) == 0 {
 		// No POs on cluster, everything is fine
-		coBuilder.WithAvailable(openshiftconfigv1.ConditionTrue, "No POs are present in the cluster", "NoPOsFound")
-		coBuilder.WithProgressing(openshiftconfigv1.ConditionTrue, "No POs are present in the cluster")
+
+		// TODO: cleanup condition reasons.
+		//   1. use constants for condition reasons.
+		//   2. discuss/agree on reason values.
+		//   3. don't use abbreviations.
+		coBuilder.WithAvailable(metav1.ConditionTrue, "NoPOsFound", "No POs are present in the cluster")
+		coBuilder.WithProgressing(metav1.ConditionTrue, "", "No POs are present in the cluster")
 		return ctrl.Result{}, nil
 	}
 
 	// check whether any of the underlying PO resources are reporting
 	// any failing status states, and update the aggregate CO resource
 	// to reflect those failing PO resources.
 	if statusErrorCheck := util.InspectPlatformOperators(poList); statusErrorCheck != nil {
-		coBuilder.WithAvailable(openshiftconfigv1.ConditionFalse, statusErrorCheck.Error(), "POError")
+		coBuilder.WithAvailable(metav1.ConditionFalse, "POError", statusErrorCheck.Error())
```

Comment:
In general should we be using the `apimachinery` api's for this instead of the `OpenShift` ones? Mainly asking to see how you decided to change this.

Reply:
They're both definitions of basically the exact same thing. I changed this mainly because apimachinery has some nice helpers for interacting with conditions that helps us avoid maintaining complexity around the "rules" of conditions, like knowing when to bump last transition time.